src/core/Switch/Switch.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Typography } from '@material-ui/core';
+import { CheckCircle, Cancel } from '@material-ui/icons';
+import { ButtonFilled } from '../Button/ButtonFilled';
+import { SwitchBaseProps } from './base';
+import { useStyles } from './styles';
+
+type Variant = 'default' | 'error' | 'success' | undefined;
+
+interface SwitchProps extends SwitchBaseProps {
+  variant?: Variant;
+}
+
+const Switch: React.FC<SwitchProps> = ({ variant }) => {
+  const classes = useStyles();
+
+  // Default Props are true
+  const [currentState, setCurrentState] = React.useState<boolean>(true);
+
+  // Toggle Handlers
+  const passToggler = () => {
+    if (currentState === false) {
+      setCurrentState(true);
+    }
+  };
+
+  const failToggler = () => {
+    if (currentState === true) {
+      setCurrentState(false);
+    }
+  };
+
+  return (
+    <div className={classes.root} data-testid="switch">
+      {/* Pass Button */}
+      <ButtonFilled
+        onClick={passToggler}
+        aria-label="left aligned"
+        variant={variant}
+        disabled={variant === 'default' || variant === 'error'}
+      >
```

Comment:
Since we are already providing the variant having an onClick here will mutate the variant. For example, if I provide a `variant="success` and click on the Second Button to trigger the failToggler then the ButtonGroup will turn into Error state while my original variant passed would still be `success`. This is one loophole